### PR TITLE
feature: add option to select a custom release

### PR DESCRIPTION
**Pull request by cpanato**
This adds support to the user select a version of cosign or use the default one

Based on the comments in the issue, it downloads the hardcoded one, does the checks, and then downloads the custom cosign, and applies the same checks, and replaces it.
Right now it is a bit hard to test with another release because just have one 😄 

Added some GitHub actions to validate the new cases.

Feel free to close this PR if does not makes sense or you don't want that.

_note_: the test actions will fail because the code does not exist in the main branch, but can check the tests here: https://github.com/cpanato/cosign-installer/runs/2218204242?check_suite_focus=true

Followup:
- make a release of this action and publish it to the Github Martkplace

/assign @dlorenc 

Fixes: https://github.com/sigstore/cosign-installer/issues/2

### CODE REVIEW

**Comment by dlorenc** [comment]
This is awesome! Thank you so much!

File: action.yml
Comment (dlorenc) [inline action.yml:39]: One nit here - I think this URL might have to change depending on the release as well, in case we rotate the key? 

Basically:

`https://raw.githubusercontent.com/sigstore/cosign/${{ inputs.cosign-release }}/.github/workflows/cosign.pub` instead of the `main`.

Comment (cpanato) [reply action.yml:39]: makes sense, thanks for your review and feedback

Comment (dlorenc) [reply action.yml]: Perfec!

**Comment by dlorenc** [comment]
This looks great to me, but I'll let @mbestavros have a look and do the merge :)

**Review by dlorenc — APPROVED** [review]

**Review by mbestavros — APPROVED** [review]
Agreed with @dlorenc -- this is exactly what we need! Code changes and tests LGTM 🙂

**Comment by cpanato** [comment]
@dlorenc @mbestavros what do y'all think to have a similar one for circleci? I will create a repo on my side and work on that and let y'all know that and if like I can transfer to this org

**Pull request merged**
